vault/auth.go

```diff
@@ -457,6 +461,9 @@ func (c *Core) teardownCredentials() error {
 // newCredentialBackend is used to create and configure a new credential backend by name
 func (c *Core) newCredentialBackend(
 	t string, sysView logical.SystemView, view logical.Storage, conf map[string]string) (logical.Backend, error) {
+	if alias, ok := credentialAliases[t]; ok {
```

Comment:
I am guessing that this upgrade code is not good enough in terms of handling the mount table entries. The `Type` of mount entries will be `aws-ec2` for the current entries and the upgrade code should change it to `aws`.

@jefferai I believe that we have run into this problem before (a long time back) and had a hard time fixing this. I _think_ it was because we had changed the auth-enable command to take in `aws-ec2` as opposed to just `aws`. Now, we need to switch back to it again :(